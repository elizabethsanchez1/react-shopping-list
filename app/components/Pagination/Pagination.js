import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Button } from 'react-native-elements'
import Icon from 'react-native-vector-icons/Entypo';
import PropTypes from 'prop-types';
import { StyledText } from '../Text';
import theme from '../../styles/theme.style';

const Pagination = ({current, total, onPress}) => {
  return (
    <View
      style={{
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
      }}
    >
      <TouchableOpacity onPress={() => onPress('previous')}>
        <Icon
          name={'chevron-left'}
          size={40}
          color={theme.ACTIVE_TAB_COLOR}
          style={{ marginRight: 40 }}
        />
      </TouchableOpacity>

      <StyledText style={{fontSize: theme.FONT_SIZE_LARGE}}>
        {current} of {total}
      </StyledText>

      <TouchableOpacity  onPress={() => onPress('next')}>
        <Icon
          name={'chevron-right'}
          size={40}
          color={theme.ACTIVE_TAB_COLOR}
          style={{ marginLeft: 40 }}
        />
      </TouchableOpacity>

    </View>
  );
};

Pagination.propTypes = {
  current: PropTypes.number,
  total: PropTypes.number,
  onPress: PropTypes.func,
};

export default Pagination;