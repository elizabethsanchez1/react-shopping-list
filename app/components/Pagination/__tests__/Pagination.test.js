import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Pagination } from "../../Pagination";

describe( 'Pagination Component', () => {

  it( 'Pagination visually looks the same', () => {
    const tree = renderer.create(
      <Pagination
        current={ 1 }
        total={ 4 }
      />
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );

} );




