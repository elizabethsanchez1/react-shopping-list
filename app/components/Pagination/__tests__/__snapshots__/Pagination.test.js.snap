// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Pagination visually looks the same 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#72DEFF",
            "fontSize": 40,
          },
          Object {
            "marginRight": 40,
          },
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "RobotoCondensed-Regular",
          "fontSize": 16,
          "fontWeight": "600",
        },
        Object {
          "fontSize": 18,
        },
      ]
    }
  >
    1
     of 
    4
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#72DEFF",
            "fontSize": 40,
          },
          Object {
            "marginLeft": 40,
          },
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
